src/connector/src/connector_common/iceberg/mod.rs

```diff
@@ -0,0 +1,595 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+mod jni_catalog;
+mod mock_catalog;
+mod storage_catalog;
+
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use anyhow::Context;
+use iceberg::io::{S3_ACCESS_KEY_ID, S3_ENDPOINT, S3_REGION, S3_SECRET_ACCESS_KEY};
+use icelake::catalog::{
+    load_iceberg_base_catalog_config, BaseCatalogConfig, CATALOG_NAME, CATALOG_TYPE,
+};
+use risingwave_common::bail;
+use serde_derive::Deserialize;
+use serde_with::serde_as;
+use url::Url;
+use with_options::WithOptions;
+
+use crate::error::ConnectorResult;
+
+#[serde_as]
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize, WithOptions)]
+pub struct IcebergCommon {
+    // Catalog type supported by iceberg, such as "storage", "rest".
+    // If not set, we use "storage" as default.
+    #[serde(rename = "catalog.type")]
+    pub catalog_type: Option<String>,
+    #[serde(rename = "s3.region")]
+    pub region: Option<String>,
+    #[serde(rename = "s3.endpoint")]
+    pub endpoint: Option<String>,
+    #[serde(rename = "s3.access.key")]
+    pub access_key: String,
+    #[serde(rename = "s3.secret.key")]
+    pub secret_key: String,
```

Comment:
referencing this part to `AwsAuthProps` instead?

Reply:
They are incompatible and this PR doesn't want to change behavior